docs/design/core/presigners/README.md

```diff
@@ -0,0 +1,525 @@
+This follows the naming strategy established by the current `{Service}Utilities` classes.
+
+##### Methods
+
+The presigner builder will have at least the following configuration:
+
+1. `region(Region)`: The region that should be used when generating the presigned URLs.
+2. `endpointOverride(URI)`: An endpoint that should be used in place of the one derived from the region.
+3. `credentialsProvider(AwsCredentialsProvider)`: The credentials that should be used when signing the request.
+
+Additional configuration will be added later after more investigation (e.g. signer, service-specific configuration).
+
+### `{Operation}PresignRequest`
+
+A new input class will be generated for each operation that supports presigning. These requests
+will extend a common base, allowing for common code to be used to configure the request.
+
+*FAQ Below: "Why a different input shape per operation?"*.
+
+#### Example
+
+```Java
+/**
+ * A request to generate presigned GetObjectRequest, passed to S3Presigner#getObject.
+ */
+public interface GetObjectPresignRequest extends PresignRequest {
+    /**
+     * The GetObjectRequest that should be presigned.
+     */
+    GetObjectRequest getObject();
+    // Plus builder boilerplate
+}
+
+public interface PresignRequest {
+    /**
+     * The duration for which this presigned request should be valid. After this time has expird,
+     * attempting to use the presigned request will fail.
+     */
+    Duration signatureDuration();
+    // Plus builder boilerplate
+}
+```
+
+### `Presigned{Operation}Request`
+
+A new output class will be generated for each operation that supports presigning. These presigned requests
+will extend a common base, allowing for common code to be used to process the response.
+
+*FAQ Below: "Why a different output shape per operation?"*.
+
+#### Example
+
+```Java
+/**
+ * A presigned GetObjectRequest, returned by S3Presigner#getObject.
+ */
+public interface PresignedGetObjectRequest extends PresignedRequest {
+    // Builder boilerplate
+}
+
+/**
+ * A generic presigned request. The isBrowserCompatible method can be used to determine whether this request
+ * can be executed by a web browser.
+ */
+public interface PresignedRequest {
+    /**
+     * The URL that the presigned request will execute against. The isBrowserCompatible method can be used to
+     * determine whether this request will work in a browser.
+     */
+    URL url();
+
+    /**
+     * The exact SERVICE time that the request will expire. After this time, attempting to execute the request
+     * will fail.
+     *
+     * This may differ from the local clock, based on the skew between the local and AWS service clocks.
+     */
+    Instant expiration();
+
+    /**
+     * Returns true if the url returned by the url method can be executed in a browser.
+     *
+     * This is true when the HTTP request method is GET, and hasSignedHeaders and hasSignedPayload are false.
+     */
+    boolean isBrowserCompatible();
```

Comment:
Hmm thoughts about renaming this? I would consider client-side JavaScript to be technically "executed in a browser" as well, in which case some of these conditions don't need to be true.

Maybe `isBrowserNavigable`?

Reply:
I agree this name isn't ideal. "Navigable" It makes it sound like the browser is a boat, though.

We could consider removing it, but I think it's important for a customer to be able to determine why their presigned URL isn't working when they hit it with a browser.

Maybe `isLinkable`?

Reply:
I like having some form of "navigation" in there since the address bar is also often referred to as the "navigation bar".

`navigableByBrowser`? :P. I think I would be fine with "linkable" as well.

Reply:
We can see what the others think.

Reply:
Updated to note that this isn't universally agreed upon as the method name and we can iterate on it in CR.